modules/imgcodecs/test/test_tiff.cpp

```diff
@@ -46,6 +48,426 @@ TEST(Imgcodecs_Tiff, decode_tile16384x16384)
     EXPECT_EQ(0, remove(file4.c_str()));
 }
 
+//==================================================================================================
+// See https://github.com/opencv/opencv/issues/22388
+
+/**
+ * Dummy enum to show combination of IMREAD_*.
+ */
+enum ImreadMixModes
+{
+    IMREAD_MIX_UNCHANGED                   = IMREAD_UNCHANGED                                     ,
+    IMREAD_MIX_GRAYSCALE                   = IMREAD_GRAYSCALE                                     ,
+    IMREAD_MIX_COLOR                       = IMREAD_COLOR                                         ,
+    IMREAD_MIX_GRAYSCALE_ANYDEPTH          = IMREAD_GRAYSCALE | IMREAD_ANYDEPTH                   ,
+    IMREAD_MIX_GRAYSCALE_ANYCOLOR          = IMREAD_GRAYSCALE                    | IMREAD_ANYCOLOR,
+    IMREAD_MIX_GRAYSCALE_ANYDEPTH_ANYCOLOR = IMREAD_GRAYSCALE | IMREAD_ANYDEPTH  | IMREAD_ANYCOLOR,
+    IMREAD_MIX_COLOR_ANYDEPTH              = IMREAD_COLOR     | IMREAD_ANYDEPTH                   ,
+    IMREAD_MIX_COLOR_ANYCOLOR              = IMREAD_COLOR                        | IMREAD_ANYCOLOR,
+    IMREAD_MIX_COLOR_ANYDEPTH_ANYCOLOR     = IMREAD_COLOR     | IMREAD_ANYDEPTH  | IMREAD_ANYCOLOR
+};
+
+/**
+ * Test lists for combination of IMREAD_*.
+ */
+const ImreadMixModes all_modes_Huge[] =
+{
+    IMREAD_MIX_UNCHANGED,
+    IMREAD_MIX_GRAYSCALE,
+    IMREAD_MIX_GRAYSCALE_ANYDEPTH,
+    IMREAD_MIX_GRAYSCALE_ANYCOLOR,
+    IMREAD_MIX_GRAYSCALE_ANYDEPTH_ANYCOLOR,
+    IMREAD_MIX_COLOR,
+    IMREAD_MIX_COLOR_ANYDEPTH,
+    IMREAD_MIX_COLOR_ANYCOLOR,
+    IMREAD_MIX_COLOR_ANYDEPTH_ANYCOLOR,
+};
+
+static inline
+void PrintTo(const ImreadMixModes& val, std::ostream* os)
+{
+    PrintTo( static_cast<ImreadModes>(val), os );
+}
+
+typedef tuple< uint64_t, tuple<string, int>, ImreadMixModes > Bufsize_and_Type;
+typedef testing::TestWithParam<Bufsize_and_Type> Imgcodecs_Tiff_decode_Huge;
+
+const uint64_t huge_buffer_sizes_decode[] =
+{
+    (uint64_t)    1 * 1024 * 1024            ,
+    (uint64_t) 1024 * 1024 * 1024 - 32768 * 4 * 2,
+    (uint64_t) 1024 * 1024 * 1024            , // 1GB
+    (uint64_t) 2048 * 1024 * 1024            , // 2GB
+};
+
+const tuple<string, int> mat_types[] =
+{
+    make_tuple("CV_8UC1",  CV_8UC1),  // 8bit  GRAY
+    make_tuple("CV_8UC3",  CV_8UC3),  // 24bit RGB
+    make_tuple("CV_8UC4",  CV_8UC4),  // 32bit RGBA
+    make_tuple("CV_16UC1", CV_16UC1), // 16bit GRAY
+    make_tuple("CV_16UC3", CV_16UC3), // 48bit RGB
+    make_tuple("CV_16UC4", CV_16UC4), // 64bit RGBA
+};
+
+TEST_P(Imgcodecs_Tiff_decode_Huge, regression)
```

Comment:
> 216 tests from Imgcodecs_Tiff/Imgcodecs_Tiff_decode_Huge (152481 ms total)

Currently these tests takes >60% of total `opencv_test_imgcodecs` time (even there are many test skips).

Default test set should check for regressions in OpenCV code.
We should not add more tests if OpenCV code coverage is not increased.

Please change:
- some small subset of tests (1-5sec in total, enabled by default)
- another "full" set of tests - mark them as `DISABLED_` (GTest special prefix). They would be used for development purposes on changing of related code.

Reply:
Thank you for your answer. Exactly the answer to the part I was having trouble with.

I split the basic tests and full tests.

After fix, basic test run at 6s with "mem_6gb" tag. It is slowly a little.

But following result is running under old PC ( i7-4770).
With recent/general pc, I think it may be more quickly.

I hope this is an acceptable result.
```
[----------] 2 tests from Imgcodecs_Tiff/Imgcodecs_Tiff_decode_Huge
[ RUN      ] Imgcodecs_Tiff/Imgcodecs_Tiff_decode_Huge.regression/0, where GetParam() = (1073479680, ("CV_8UC1", 0), IMREAD_COLOR)
[       OK ] Imgcodecs_Tiff/Imgcodecs_Tiff_decode_Huge.regression/0 (3317 ms)
[ RUN      ] Imgcodecs_Tiff/Imgcodecs_Tiff_decode_Huge.regression/1, where GetParam() = (2147483648, ("CV_16UC4", 26), IMREAD_COLOR)
[       OK ] Imgcodecs_Tiff/Imgcodecs_Tiff_decode_Huge.regression/1 (3191 ms)
[----------] 2 tests from Imgcodecs_Tiff/Imgcodecs_Tiff_decode_Huge (6511 ms total)
```